src/include/execution/compiler/translator_factory.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <memory>
+#include "execution/compiler/expression/expression_translator.h"
+#include "execution/compiler/operator/operator_translator.h"
+#include "execution/util/region.h"
+#include "parser/expression/abstract_expression.h"
+#include "planner/plannodes/abstract_plan_node.h"
+namespace terrier::execution::compiler {
+
+class Pipeline;
+class OperatorTranslator;
+
+/**
+ * Translator Factory
+ */
+class TranslatorFactory {
+ public:
+  /**
+   * Create a regular expression translator
+   */
+  static std::unique_ptr<OperatorTranslator> CreateRegularTranslator(const terrier::planner::AbstractPlanNode *op,
+                                                                     CodeGen *codegen);
+
+  /**
+   * Create a bottom expression translator
+   */
+  static std::unique_ptr<OperatorTranslator> CreateBottomTranslator(const terrier::planner::AbstractPlanNode *op,
+                                                                    CodeGen *codegen);
+
+  /**
+   * Create a top expression translator
+   */
+  static std::unique_ptr<OperatorTranslator> CreateTopTranslator(const terrier::planner::AbstractPlanNode *op,
+                                                                 OperatorTranslator *bottom, CodeGen *codegen);
+
+  /**
+   * Create a left expression translator
+   */
+  static std::unique_ptr<OperatorTranslator> CreateLeftTranslator(const terrier::planner::AbstractPlanNode *op,
+                                                                  CodeGen *codegen);
+
+  /**
+   * Create a right expression translator
+   */
+  static std::unique_ptr<OperatorTranslator> CreateRightTranslator(const terrier::planner::AbstractPlanNode *op,
+                                                                   OperatorTranslator *left, CodeGen *codegen);
+
+  /**
+   * Creates an expression translator
+   */
+  static std::unique_ptr<ExpressionTranslator> CreateExpressionTranslator(
+      const terrier::parser::AbstractExpression *expression, CodeGen *codegen);
+
+  /**
+   * Whether this is a comparison operation
+   */
+  static bool IsComparisonOp(terrier::parser::ExpressionType type) {
+    return ((type) <= terrier::parser::ExpressionType::COMPARE_IS_DISTINCT_FROM &&
+            (type) >= terrier::parser::ExpressionType::COMPARE_EQUAL);
```

Comment:
Using range comparisons seems pretty brittle given we may add operators in the future (applies below as well).  It may be more verbose, but I think it is more maintainable to do a long series of equality comparisons (switch statement?) since it does not rely on enum layout.  A good compiler should consolidate the ranges anyway.